pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -445,6 +446,23 @@ func (p *criStatsProvider) getFsInfo(fsID *runtimeapi.FilesystemIdentifier) *cad
 	mountpoint := fsID.GetMountpoint()
 	fsInfo, err := p.cadvisor.GetDirFsInfo(mountpoint)
 	if err != nil {
+		available, capacity, usage, inodes, inodesFree, _, duErr := fs.Info(mountpoint)
+		if duErr == nil {
+			fsInodes := uint64(inodes)
+			fsInodesFree := uint64(inodesFree)
+			return &cadvisorapiv2.FsInfo{
+				Timestamp:  time.Now(),
+				Mountpoint: mountpoint,
+				Capacity:   uint64(capacity),
+				Available:  uint64(available),
+				Usage:      uint64(usage),
+				Inodes:     &fsInodes,
+				InodesFree: &fsInodesFree,
+			}
+		} else {
+			klog.ErrorS(duErr, "Failed to get filesystem info", "mountpoint", mountpoint)
+		}
+
```

Comment:
Would it make sense to do in cadvisor.GetDirFsInfo code?

Reply:
I used  "k8s.io/kubernetes/pkg/volume/util/fs" `fs.Info(mountpoint)` here.

If we implement this in cadvisor, there may be a dup function like `fs.Info(mountpoint)` using `unix.Statfs` to do a similar thing.

> Would it make sense to do in cadvisor.GetDirFsInfo code?

 @bobbypage @dashpole  do you agree? Is this a cadvisor bug or not?

Reply:
@pacoxu I was suggesting to extend cadvisor.GetDirFsInfo to handle this error instead of handling it on the caller side.

Reply:
I opened https://github.com/google/cadvisor/issues/3234  and I will submit a PR later in cadvisor to fix it then.
/hold
maybe we can close this pr after I submit a pr in cadvisor.

Reply:
@pacoxu any news on this? Should we close this PR?

Reply:
/close
but I still have no enough time to work on a pr in cadvisor to address it.😓